File: DPLABFINAL/DPLabFinal/src/UpdateIssueCommand.java
public class UpdateIssueCommand implements Command{
    private IssueRepository repository;
    private Issue issue;
    private String newTitle;
    private String newDescription;

    public UpdateIssueCommand(IssueRepository repository, Issue issue, String newTitle, String newDescription) {
        this.repository = repository;
        this.issue = issue;
        this.newTitle = newTitle;
        this.newDescription = newDescription;
    }

    @Override
    public void execute()
    {
        issue.setTitle(newTitle);
        issue.setDescription((newDescription));
    }

    @Override
    public void undo()
    {
        System.out.println("tuli undo the update here!!");
    }

}

File: TaskManager/TaskManager/TaskManager/Commands/UpdateIssueCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using TaskManager.Models;
using TaskManager.Repository;
using System.Threading.Tasks;

namespace TaskManager.Commands
{
    namespace TaskManager.Commands
    {
        public class UpdateIssueCommand : ICommand
        {
            private readonly IssueRepository _repository;
            private readonly Issue _newIssue;
            private readonly Issue _oldIssue;

            public UpdateIssueCommand(IssueRepository repository, Issue newIssue)
            {
                _repository = repository;
                _newIssue = newIssue;
                _oldIssue = repository.GetById(newIssue.Id);
                Timestamp = DateTime.UtcNow;
            }

            public string Description => $"Updated issue: {_newIssue.Title}";
            public DateTime Timestamp { get; }
            public CommandType Type => CommandType.Update;

            public void Execute()
            {
                _repository.Update(_newIssue);
            }

            public void Undo()
            {
                _repository.Update(_oldIssue);
            }
        }
    }
}
